File: 0011 盛最多水的容器/main.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        li=0
        ri=len(height)-1
        maxWater=0
        while(li!=ri):
            width = ri-li
            deepth = min(height[li],height[ri])
            V = width*deepth
            if(V>maxWater):
                maxWater=V
            if(height[li]>height[ri]):
                ri-=1
            else:
                li+=1
        return maxWater

File: 0011 盛最多水的容器/main.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& height) {
        auto li=height.begin(),ri=height.end()-1;
        int maxSize = min(*li,*ri)*(ri-li);
        int size=0;
        int hli=0,hri=0;
        while(ri>li){
            hri=*ri;
            hli=*li;
            if( hli  > hri){

                while( ri> li && *--ri > hri ){
                }
                size=min(*li,*ri)*(ri-li);
                if(size>maxSize){
                    maxSize=size;
                }
            }
            else{
                while( ri> li && *++li > hli ){
                    cout<< "2:" << *li;
                }
                size=min(*li,*ri)*(ri-li);
                if(size > maxSize){
                    maxSize=size;
                }
            }
        }
        return maxSize;
    }
};